### PR TITLE
Policy propriatary telemetry policy handler

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**

### CODE REVIEW

**Comment by LuxoftAKutsan** [comment]
@AGritsevich @LevchenkoS  please look throw

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,7 @@
 build/
-*.user
+*.user*
```
Comment (AGritsevich) [inline .gitignore]: @LuxoftAKutsan , @LevchenkoS It's  for QtCreator, will stay it?

**Comment by LuxoftAKutsan** [comment]
Review will be finished within #1069

File: .gitignore (hunk 1 above)
Comment (LuxoftAKutsan) [inline .gitignore]: @AGritsevich yes, to see clear git. As you know moust of the developers use qtcreator

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
 src/appMain/sdl_preloaded_pt.json
+CMakeCache.txt
+CMakeFiles/
+tools/Utils/c_rehash.pyc
+tools/infrastructure/utils.pyc
```
Comment (LuxoftAKutsan) [inline .gitignore]: @AGritsevich i can see in latest version ```*.pyc¬ ``` that is correct

File: src/components/application_manager/include/application_manager/application_impl.h [hunk 3]
```diff
@@ -156,7 +156,7 @@ class ApplicationImpl : public virtual InitialApplicationDataImpl,
   void set_device(connection_handler::DeviceHandle device);
   virtual uint32_t get_grammar_id() const;
   virtual void set_grammar_id(uint32_t value);
-
+  bool is_audio() const OVERRIDE;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/application_impl.h]: @AGritsevich I don;t think so, looks like rather common interface

File: src/components/application_manager/include/application_manager/policies/delegates/app_permission_delegate.h [hunk 4]
```diff
@@ -1,82 +1,5 @@
...
-};
-
-}  // namespace policy
-
-#endif  // SRC_COMPONENTS_APPLICATION_MANAGER_INCLUDE_APPLICATION_MANAGER_DELEGATES_APP_PERMISSION_DELEGATE_H_
+#ifdef EXTENDED_PROPRIETARY
+#include "application_manager/policies/premium/delegates/app_permission_delegate.h"
+#else
+#include "application_manager/policies/regular/delegates/app_permission_delegate.h"
+#endif
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/policies/delegates/app_permission_delegate.h]: @AGritsevich this fixed

File: src/components/application_manager/include/application_manager/application_manager.h [hunk 5]
```diff
@@ -48,7 +48,7 @@
 #include "application_manager/application_manager_settings.h"
 #include "application_manager/state_controller.h"
 #include "application_manager/hmi_interfaces.h"
-
+#include "policy/policy_types.h"
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/application_manager.h]: @AGritsevich looks like common include

File: src/components/application_manager/include/application_manager/policies/policy_handler.h [hunk 6]
```diff
@@ -466,30 +483,32 @@ class PolicyHandler : public PolicyHandlerInterface,
       policy_handler_->AsyncRun(new StatisticsDelegate(*policy_handler_, type));
     }
 
-    void Increment(const std::string& app_id,
-                   usage_statistics::AppCounterId type) {
+    virtual void Increment(const std::string& app_id,
+                           usage_statistics::AppCounterId type) {
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/policies/policy_handler.h]: @AGritsevich  I will

File: src/components/application_manager/include/application_manager/policies/premium/delegates/app_permission_delegate.h [hunk 7]
```diff
@@ -0,0 +1,83 @@
+﻿/*
+ Copyright (c) 2014, Ford Motor Company
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/policies/premium/delegates/app_permission_delegate.h]: @AGritsevich yes cc1dff8

**Pull request closed without merging**